File: src/main/java/buttondevteam/discordplugin/commands/VersionCommand.java
package buttondevteam.discordplugin.commands;

import buttondevteam.discordplugin.DiscordPlugin;
import buttondevteam.lib.chat.CommandClass;
import lombok.val;

@CommandClass(helpText = {
	"Version",
	"Returns the plugin's version"
})
public class VersionCommand extends ICommand2DC {
	public boolean def(Command2DCSender sender) {
		sender.sendMessage(getVersion());
		return true;
	}

	public static String[] getVersion() {
		val desc = DiscordPlugin.plugin.getDescription();
		return new String[]{ //
				desc.getFullName(), //
				desc.getWebsite() //
		};
	}
}

File: src/main/scala/buttondevteam/discordplugin/commands/VersionCommand.scala
package buttondevteam.discordplugin.commands

import buttondevteam.discordplugin.DiscordPlugin
import buttondevteam.lib.chat.{Command2, CommandClass}

@CommandClass(helpText = Array("Version", "Returns the plugin's version"))
object VersionCommand {
    def getVersion: Array[String] = {
        val desc = DiscordPlugin.plugin.getDescription
        Array[String](desc.getFullName, desc.getWebsite)
    }
}

@CommandClass(helpText = Array("Version", "Returns the plugin's version"))
class VersionCommand extends ICommand2DC {
    @Command2.Subcommand override def `def`(sender: Command2DCSender): Boolean = {
        sender.sendMessage(VersionCommand.getVersion)
        true
    }
}